src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaRepository.java

```diff
@@ -41,42 +45,68 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(KafkaRepository.class);
 
-    private static final long KAFKA_SEND_TIMEOUT_MS = 10000;
-
-    private static final long KAFKA_READ_TIMEOUT_MS = 0;
-
     private final ZooKeeperHolder zkFactory;
     private final Producer<String, String> kafkaProducer;
     private final KafkaFactory kafkaFactory;
-
-    @Value("${nakadi.kafka.poll.timeoutMs}")
-    private long kafkaPollTimeout;
+    private final KafkaRepositorySettings settings;
 
     @Autowired
-    public KafkaRepository(final ZooKeeperHolder zkFactory, final KafkaFactory kafkaFactory) {
+    public KafkaRepository(final ZooKeeperHolder zkFactory, final KafkaFactory kafkaFactory,
+                           final KafkaRepositorySettings settings) {
         this.zkFactory = zkFactory;
         this.kafkaProducer = kafkaFactory.createProducer();
         this.kafkaFactory = kafkaFactory;
+        this.settings = settings;
     }
 
     @Override
     public List<Topic> listTopics() throws NakadiException {
         try {
-            return zkFactory.get().getChildren().forPath("/brokers/topics").stream().map(s ->
-                    new Topic(s)).collect(Collectors.toList());
+            return zkFactory.get()
+                    .getChildren()
+                    .forPath("/brokers/topics")
+                    .stream()
+                    .map(Topic::new)
+                    .collect(Collectors.toList());
         } catch (Exception e) {
             throw new NakadiException("Failed to get partitions", e);
         }
     }
 
     @Override
+    public void createTopic(final String topic) {
+        createTopic(topic,
+                settings.getDefaultTopicPartitionNum(),
+                settings.getDefaultTopicReplicaFactor(),
+                settings.getDefaultTopicRetentionMs(),
+                settings.getDefaultTopicRotationMs());
+    }
+
+    @Override
+    public void createTopic(final String topic, final int partitionsNum, final int replicaFactor,
+                            final long retentionMs, final long rotationMs) {
+
+        final String connectionString = zkFactory.get().getZookeeperClient().getCurrentConnectionString();
+        final ZkClient zkClient = new ZkClient(connectionString, settings.getZkSessionTimeoutMs(),
+                settings.getZkConnectionTimeoutMs(), ZKStringSerializer$.MODULE$);
+        final ZkUtils zkUtils = ZkUtils.apply(zkClient, false);
+
+        final Properties topicConfig = new Properties();
+        topicConfig.setProperty("retention.ms", Long.toString(retentionMs));
+        topicConfig.setProperty("segment.ms", Long.toString(rotationMs));
+
+        AdminUtils.createTopic(zkUtils, topic, partitionsNum, replicaFactor, topicConfig);
+        zkClient.close();
```

Comment:
I was thinking if is there any chance of an error causing this connection to remain open. Do you think we should wrap this statement in a `finally` clause?